controller/Review.php

```diff
@@ -69,26 +79,37 @@ public function __construct()
      * @throws LtiVariableMissingException
      * @throws common_exception_Error
      * @throws common_exception_NotFound
+     * @throws common_Exception
      */
     public function index(): void
     {
         $launchData = $this->ltiSession->getLaunchData();
-
-        /** @var DeliveryExecutionFinderService $finder */
-        $finder = $this->getServiceLocator()->get(DeliveryExecutionFinderService::SERVICE_ID);
-
-        $execution = $finder->findDeliveryExecution($launchData);
+        $finder = $this->getDeliveryExecutionFinderService();
+        $deliveryId = $this->getDeliveryId();
+
+        if ($deliveryId === null) {
```

Comment:
I think the validation here should be a bit different.
Let's ensure `delivery` is there for cases when `$launchData->getVariable(LtiLaunchData::LTI_MESSAGE_TYPE) === LtiMessageInterface::LTI_MESSAGE_TYPE_SUBMISSION_REVIEW_REQUEST)`. If `delivery` query parameter was not provided, and `LtiClientException` should be thrown.

Reply:
and if message type is not LTI_MESSAGE_TYPE_SUBMISSION_REVIEW_REQUEST, should check is delivery execution id provided in a claim?

Reply:
I think it's already covered by `findDelvieryExecution()` implementation
```php
        if ($deliveryExecutionId === null) {
            throw new LtiInvalidLaunchDataException('Wrong result ID provided');
        }
```

Reply:
something like [that](https://github.com/oat-sa/extension-lti-test-review/pull/92/files#diff-245bb8f839d90cef1105315ff9ed71c502fc9fe6e43c374ca83744237077d635R89)?